api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/Conv1DTranspose.kt

```diff
@@ -0,0 +1,103 @@
+ *
+ * This layer expects input data of size `(N, L, C)` where
+ * ```
+ * N - batch size
+ * L - length of signal sequence
+ * C - number of channels
+ * ```
+ *
+ * Note: dilation values greater than 1 are not supported on cpu
+ * (see https://github.com/tensorflow/tensorflow/issues/28264).
+ *
+ * @property [filters] dimensionality of the output space (i.e. the number of filters in the convolution)
+ * @property [kernelLength] size of the convolutional kernel (one number)
+ * @property [strides] strides of the convolution for each dimension of the input tensor (three numbers)
+ * @property [dilations] dilations of the convolution for each dimension of the input tensor (three numbers).
+ *           Currently, dilation values greater than 1 are not supported on cpu.
+ * @property [activation] activation function
+ * @property [kernelInitializer] initializer for the kernel
+ * @property [biasInitializer] initializer for the bias
+ * @property [kernelRegularizer] regularizer for the kernel
+ * @property [biasRegularizer] regularizer for the bias
+ * @property [activityRegularizer] regularizer function applied to the output of the layer
+ * @property [padding] type of padding to use
+ * @property [outputPadding] the amount of explicit padding to use (six numbers: two for each dimension).
+ * @property [useBias] a flag that specifies if the bias should be used
+ * @param [name] custom layer name
+ */
+public class Conv1DTranspose(
+    public override val filters: Int = 3,
+    public val kernelLength: Int = 3,
+    public override val strides: IntArray = intArrayOf(1, 1, 1),
+    public override val dilations: IntArray = intArrayOf(1, 1, 1),
+    public override val activation: Activations = Activations.Relu,
+    public override val kernelInitializer: Initializer = HeNormal(),
+    public override val biasInitializer: Initializer = HeUniform(),
+    public override val kernelRegularizer: Regularizer? = null,
+    public override val biasRegularizer: Regularizer? = null,
+    public override val activityRegularizer: Regularizer? = null,
+    public override val padding: ConvPadding = ConvPadding.SAME,
+    public override val outputPadding: IntArray? = null,
+    public override val useBias: Boolean = true,
+    name: String = ""
+) : ConvTranspose(dimensions = 1, name), NoGradients {
+
+    init {
+        requireArraySize(strides, dimensions + 2, "strides")
+        requireArraySize(dilations, dimensions + 2, "dilations")
+        if (outputPadding != null) requireArraySize(outputPadding, 2 * (dimensions + 2), "outputPadding")
+        isTrainable = false
+    }
+
+    override val kernelSize: IntArray = intArrayOf(kernelLength)
+
+    override fun convImplementation(tf: Ops, input: Operand<Float>): Operand<Float> {
+        return tf.withExpandedDimensions(input) { expandedInput ->
+            val expandedOutputPadding = outputPadding?.withAdded(EXTRA_DIM * 2, listOf(0, 0))
+            return@withExpandedDimensions tf.nn.conv2dBackpropInput(
```

Comment:
very complex function call, could we extract some logic to the separate variables and leave some comments for the future (why, for example, we did` tf.shapeWithDynamicBatchSize(outputShape.expand(), input)` for the specific parameter and so on) At this moment this knowledge is hidden in the issue comments.
I suggest doing this, because it's enough far from the initial Keras code

Reply:
> why, for example, we did tf.shapeWithDynamicBatchSize(outputShape.expand(), input)

`shapeWithDynamicBatchSize` has a javadoc with explanation and links to the relevant issues, I'll add some comments to this method as well.